documentation/src/docs/pravega-concepts.md

```diff
@@ -93,6 +114,13 @@ Stream Segments. Event Routing Keys are hashed to form a "key space". The key
 space is then divided into a number of partitions, corresponding to the number
 of Stream Segments. Consistent hashing determines of Events to Stream Segments.
 
+### Stream Segments and Connection pooling
+
+When an Event is written to the Stream by the Pravega Client it is written into one of the Stream Segments based on the Event Routing Key. These Segments which are a part of Segment Containers are managed by the different [Segment Store](segment-store-service.md) Service instances.
+
+At present the Pravega creates new connections to different Segment Stores for every Segment it is writing to. A new connection to a Segment Store is created even when multiple Segments are owned by the same Segment Store. Every Segment being read by the Pravega client maps to a new connection.
```

Comment:
Why do you describe historical data in a documentation file? This would work in a blog post. The documentation should present the current way things work.

Reply:
@andreipaduroiu Accepted suggestion